src/components/progress-indicators/StepProgressBar.js

```diff
@@ -92,8 +92,10 @@ const StepProgressBarComponent = (props: { actionDisplayInfos: ActionDisplayInfo
       // Active/in-progress nodes are partially filled while queued or completed
       // nodes are solid filled.
       const isLast = totalSteps <= 1 || i >= totalSteps - 1
-      // HACK: Also set active status of this node based on the status of the previous node. Check to be removed when ActionQueue properly handles updating its execution state.
-      const isNodeCompleted = actionDisplayInfos[i].status === 'done' || actionDisplayInfos[i].status instanceof Error
+      // HACK: Also set the completed state of this node based on the status of the next node. Check to be removed when ActionQueue has the ability to store the state of a particular step as it's happening or after implementing a way to reliably infer the status of a step from the (post-execution) effect.
+      const isNodeCompleted =
+        actionDisplayInfos[i].status === 'done' || actionDisplayInfos[i].status instanceof Error || (!isLast && actionDisplayInfos[i + 1].status !== 'pending')
```

Comment:
How the GUI displays the state of the ActionProgram != the design of ActionQueue. TBD post-aave-beta.

Reply:
We spoke before about a temporary workaround here about deriving the state of the program purely from the ActionOp effects. This change is allowing it to work without investigating whether that's even possible

Reply:
Test if this is needed. If it's not determined to be necessary, let's save face by not adding it because the cost to maintaining the codebase increase.

Reply:
Conditions were removed here after testing